src/main/java/bio/terra/service/filedata/azure/tables/TableFileDao.java

```diff
@@ -0,0 +1,160 @@
+package bio.terra.service.filedata.azure.tables;
+
+import bio.terra.service.filedata.exception.FileSystemCorruptException;
+import bio.terra.service.filedata.exception.FileSystemExecutionException;
+import bio.terra.service.filedata.google.firestore.ApiFutureGenerator;
+import bio.terra.service.filedata.google.firestore.FireStoreDirectoryEntry;
+import bio.terra.service.filedata.google.firestore.FireStoreFile;
+import bio.terra.service.filedata.google.firestore.InterruptibleConsumer;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.tables.TableClient;
+import com.azure.data.tables.TableServiceClient;
+import com.azure.data.tables.models.ListEntitiesOptions;
+import com.azure.data.tables.models.TableEntity;
+import com.azure.data.tables.models.TableServiceException;
+import com.google.api.core.SettableApiFuture;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+/**
+ * TableFileDao provides CRUD operations on the "files" storage table in Azure. Objects in the file
+ * table are referred to by the dataset (owner of the files) and by any snapshots that reference the
+ * files. File naming is handled by the directory DAO. This DAO just handles the basic operations
+ * for managing the collection. It does not have logic for protecting against deleting files that
+ * have dependencies or creating files with duplicate paths.
+ */
+@Component
+public class TableFileDao {
+  private final Logger logger = LoggerFactory.getLogger(TableFileDao.class);
+  private final ExecutorService executor;
+  private static final String TABLE_NAME = "files";
+  private static final String PARTITION_KEY = "partitionKey";
+  private static final int SLEEP_BASE_SECONDS = 1;
+  private static final int MAX_SLEEP_SECONDS = 300;
+  private static final int AZURE_STORAGE_RETRIES = 1;
```

Comment:
Will we want to move any of these to application.properties?

Reply:
That's a good point! I'll do that in the PR I have for the file directory dao since I'll need to move/re-use some of the methods here

Reply:
Sounds good!